climpred/graphics.py

```diff
@@ -230,3 +236,219 @@ def plot_bootstrapped_skill_over_leadyear(bootstrapped, plot_persistence=True, a
     ax.legend(frameon=False)
     ax.set_xlabel('Lead time [years]')
     return ax
+
+
+def _check_only_climpred_dims(pe):
+    """Warns if dimensions other than `CLIMPRED_DIMS` are in `PredictionEnsemble`."""
+    additional_dims = set(pe.get_initialized().dims) - set(CLIMPRED_DIMS)
+    if len(additional_dims) != 0:
+        raise DimensionError(
+            f'{type(pe.__name__)}.plot() does not allow dimensions other '
+            f'than {CLIMPRED_DIMS}, found {additional_dims}. '
+            f'Please use .mean({additional_dims}) '
+            f'or .isel() before plot.'
+        )
+
+
+def plot_lead_timeseries_hindcast(
+    he, variable=None, ax=None, show_members=False, cmap='jet'
+):
+    """Plot datasets from HindcastEnsemble.
+
+    Args:
+        he (HindcastEnsemble): HindcastEnsemble.
+        variable (str or None): `variable` to plot. Defaults to the first in data_vars.
+        ax (plt.axes): Axis to use in plotting. By default, creates a new axis.
+        show_members (bool): whether to display all members individually.
+            Defaults to True.
+        cmap (str): Name of matplotlib-recognized colorbar. Defaults to 'jet'.
+
+    Returns:
+        ax: plt.axes
+
+    """
+    _check_only_climpred_dims(he)
+    if not variable:
+        variable = list(he.get_initialized().data_vars)[0]
+    hind = he.get_initialized()[variable]
+    lead_freq = get_lead_cftime_shift_args(hind.lead.attrs['units'], 1)[1]
+    hist = he.get_uninitialized()
+    if isinstance(hist, xr.Dataset):
+        hist = hist[variable]
+    obs = he._datasets['observations']
+
+    cmap = mpl.cm.get_cmap(cmap, hind.lead.size)
+    if not ax:
+        _, ax = plt.subplots(figsize=(10, 4))
+    if isinstance(hist, xr.DataArray):
+        if 'member' in hist.dims and not show_members:
+            hist = hist.mean('member')
+            member_alpha = 1
+            lw = 2
+        else:
+            member_alpha = 0.4
+            lw = 1
+        hist.plot(
+            ax=ax,
+            lw=lw,
+            hue='member',
+            color='gray',
+            alpha=member_alpha,
+            label='uninitialized',
+            zorder=hind.lead.size + 1,
+        )
+
+    for i, lead in enumerate(hind.lead.values):
+        h = hind.sel(lead=lead).rename({'init': 'time'})
+        if not show_members and 'member' in h.dims:
+            h = h.mean('member')
+            lead_alpha = 1
+        else:
+            lead_alpha = 0.5
+        h['time'] = shift_cftime_index(h.time, 'time', int(lead), lead_freq)
+        h.plot(
+            ax=ax,
+            hue='member',
+            color=cmap(i),
+            label=f'initialized: lead={lead}',
+            alpha=lead_alpha,
+            zorder=hind.lead.size - i,
+        )
+
+    linestyles = ['-', ':', '-.', '--']
+    if len(obs) > len(linestyles):
+        raise ValueError(f'Please provide fewer than {len(linestyles)} observations.')
```

Comment:
I think this should be `{len(linestyles)+1}` but you could pick that up in another PR so we don't have to rerun this testing.

Reply:
I'll quickly fix that

Reply:
If you change the other things here you can change this too